src/main/java/com/opentable/versionedconfig/MutableUri.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.opentable.versionedconfig;
+
+import java.net.URI;
+import java.util.Objects;
+
+import org.springframework.web.util.UriComponentsBuilder;
+
+public class MutableUri implements Comparable<MutableUri> {
+    private final String host;
+    private final int port;
+    private final String path;
+    private String username;
+    private String password;
+    private String scheme;
+    public MutableUri(URI uri) {
+        this.host = uri.getHost();
+        this.port = uri.getPort();
+        this.path = uri.getPath();
+        this.scheme = uri.getScheme();
+        if (uri.getUserInfo() != null) {
+            String[] userInfo = uri.getUserInfo().split(":");
```

Comment:
should check userInfo for length, to prevent NPE

Reply:
done